coffea/nanoevents/factory.py

```diff
@@ -329,15 +344,19 @@ def events(self):
         """Build events"""
         events = self._events()
         if events is None:
+            prefix = self._partition_key
+
+            def key_formatter(partition, form_key, attribute):
+                return prefix + f'/{partition}/{form_key}/{attribute}'
+
             behavior = dict(self._schema.behavior)
             behavior["__events_factory__"] = self
-            events = awkward1.from_arrayset(
+            events = ak.from_buffers(
                 self._schema.form,
+                len(self),
                 self._mapping,
-                prefix=self._partition_key,
-                sep="/",
+                key_format=key_formatter,
```

Comment:
I missed this development, thanks @jpivarski ! Does this allow also control of the non-content attributes such as `offsets`? That would simplify the DSL a bit by removing the need for the `!skip` command (because otherwise awkward1 forces us to add an offsets that is sometimes not appropriate (e.g. if the offsets are coming from a different branch)

Reply:
Indeed it does, per docs:
key_format (str or callable) – Python format string containing "{partition}", "{form_key}", and/or "{attribute}" or a function that takes these as keyword arguments and returns a string to use as keys for buffers in the container. The partition is a partition number (non-negative integer, passed as a string), the form_key is a string associated with each node in the Form, and the attribute is a hard-coded string representing the buffer’s function (e.g. "data", "offsets", "index").

Reply:
So the problem here is that `/offsets` was generated by the old `from_arrayset` and the others are new. Dropping the attribute results in an incorrectly unpacked array. It may be better to account for the new attributes and work with @jpivarski to tune in a well defined and useful behavior.

Reply:
delaying this to rc2